utilities/transactions/write_unprepared_txn_db.cc

```diff
@@ -429,7 +428,9 @@ Iterator* WriteUnpreparedTxnDB::NewIterator(const ReadOptions& options,
   // For keys that were not modified by this transaction, largest_validated_seq_
   // is meaningless, and Prev() should just work with the existing visibility
   // logic.
-  if (txn->largest_validated_seq_ > snapshot->GetSequenceNumber()) {
+  if (txn->largest_validated_seq_ > snapshot->GetSequenceNumber() &&
+      !txn->unprep_seqs_.empty()) {
+    ROCKS_LOG_ERROR(info_log_, "WriteUnprepared iterator creation failed");
```

Comment:
Can we add a more descriptive comment? like "WriteUnprepared iterator creation failed since there are the transaction has performed unvalidated writes."